### PR TITLE
Update RAC recipe

### PR DESCRIPTION
**Pull request by leonheldattoradex**
This release of RAC includes the possibility of receive messages through D-Bus to poll the server for new sessions. These messages are sent by the tzn-mqtt bridge (now included as a dependency for the remote-access service) and thus we can significantly increase the intervals to check the HTTP server, reducing bandwidth.

Still, we keep the periodically checking the HTTP server regardless of new messages on D-Bus as a fallback solution.

This default configuration is an example and can be changed by users to fit their particular needs, specially in the case where an applicaiton is bandwidth sensitive.

Note that this change also moves the rac repository to the Torizon organization and also slightly updates the comment in regards to AUTOREV.

Related-to: TOR-3522

This change was tested with `qemuarm64` and production deployment of Torizon Cloud with the following command

```
qemu-system-aarch64 \
    -M virt \
    -cpu cortex-a72 \
    -m 2048 \
    -nographic \
    -drive if=none,file=torizon-docker-qemuarm64.wic,id=hd0,format=raw \
    -device virtio-blk-device,drive=hd0 \
    -device virtio-net-device,netdev=net0 \
    -netdev user,id=net0,hostfwd=tcp::2223-:22 \
    -bios u-boot.bin
```

and the usual provisioning steps plus `sudo systemctl enable remote-access && sudo systemctl restart remote-access`.

### CODE REVIEW

**Review by jsrc27 — DISMISSED** [review]
LGTM.

We probably want another PR to include this change in master right?

File: recipes-sota/rac/files/client.toml
Comment (leonheldattoradex) [reply recipes-sota/rac/files/client.toml:13]: > does it mean it is always performed more like a secondary method independent of MQTT?

This. It's performed in a fixed interval (set by `poll_timeout`) so it doesn't check if the MQTT connection is up or anything, just whether it hasn't received any MQTT messages via D-Bus for `poll_timeout` seconds.
The `validation_poll_timeout` is the same logic but for the validation of the session after it has already been established.

> If the polling is always performed even when the MQTT connection is up, I'd ask if this polling interval is appropriate as a default. I imagine the setting here will represent a least a couple of megabytes/month/device, right?

Yes, you're correct. It's miles better than it was before (polling every 3s, which was very bandwidth-heavy) but may still be too much for some users. My original intention was to completely disable the HTTP polling by adding a new option `event_mode_only`, but that got scraped - for now - after a meeting with the cloud team. I think having this longer poll time until we land those features and the MQTT feature stabilizes in the cloud is a good strategy.

Let me know what you think!

File: recipes-sota/rac/files/remote-access.service
Comment (leonheldattoradex) [reply recipes-sota/rac/files/remote-access.service:4]: > Is tzn-mqtt.service a strict requirement of rac? 

No and...

>  If rac can work even in some "fallback" mode, then it's better to use a Wants here...

Yes, you're correct. I'll make the changes.

File: recipes-sota/rac/files/client.toml
Comment (EdTheBearded) [reply recipes-sota/rac/files/client.toml:13]: Is is possible to verify if MQTT is working properly to actually avoid this fallback? If so, this could become an improvement ticket, since it seems we still have the problem we were trying to solve initially.

Comment (leograba) [reply recipes-sota/rac/files/client.toml:13]: We know that data usage is a concern.

A suggestion:

* Use a reasonable default value, and implement it in a way it can easily be overridden in /etc so users can capture changes with TCB
* Create a section on https://developer.toradex.com/torizon/torizon-platform/remote-access/ talking about data usage, how to tweak default values, and what are the trade-offs between longer polling time and service availability

I'm just jumping in because I feel like this usage requirement may impact the implementation, so rather sooner than later to avoid rework, etc.

**Comment by EdTheBearded** [comment]
> LGTM.
> 
> We probably want another PR to include this change in master right?

@leonheldattoradex could you please also submit a PR for master branch?

File: recipes-sota/rac/files/client.toml
Comment (leonheldattoradex) [reply recipes-sota/rac/files/client.toml:13]: > Is is possible to verify if MQTT is working properly to actually avoid this fallback?

Actually, not easily because of the architecture. The code that would be responsible for this check would be `tzn-mqtt`, not `rac`. So there would need to maybe be a new message from `tzn-mqtt` basically saying "cannot reach MQTT broker, can you please use HTTP k thx bye" to `rac` (and `aktualizr` for that matter).

Comment (EdTheBearded) [reply recipes-sota/rac/files/client.toml:13]: Gotcha. If it possible to do, I think it would be nice to have this improvement captured in a ticket, and meanwhile we have this "fallback always on" approach.

Comment (leonheldattoradex) [reply recipes-sota/rac/files/client.toml:13]: @leograba I guess this is more of a decision for you then: should we do this work now or merge as-is - with the corrections maintainers asked for, naturally - (which I'm in favor of because it's a beta feature and needs testing and it's incrementally better than the previous state)?

Comment (leograba) [reply recipes-sota/rac/files/client.toml:13]: Yes, merge as-is sounds good. No need to hold improvements

Comment (leonheldattoradex) [reply recipes-sota/rac/files/client.toml:13]: Ok, I'll capture the improvements in tickets.

**Comment by leonheldattoradex** [comment]
> > > LGTM.
> > > We probably want another PR to include this change in master right?
> > 
> > 
> > @leonheldattoradex could you please also submit a PR for master branch?
> 
> Yes, but I'll wait for the discussions here to come to a close, is that ok?

https://github.com/torizon/meta-toradex-torizon/pull/154

**Review by rborn-tx — APPROVED** [review]
LGTM

**Review by EdTheBearded — APPROVED** [review]

**Pull request merged**
